packages/react/src/components/form/authenticating/otp.tsx

```diff
@@ -42,17 +42,21 @@ const FactorIcon = ({ factor }: { factor: Factor }) => {
 export const OTPState = ({ flowState }: Props) => {
   const { text } = useConfiguration();
   const { sid } = useSlashID();
-  const { values, registerField, registerSubmit } = useForm();
+  const {
+    values,
+    registerField,
+    registerSubmit,
+    setError,
+    hasError,
+    clearError,
+  } = useForm();
   const [formState, setFormState] = useState<
-    "initial" | "input" | "submitting"
+    "initial" | "input" | "submitting" | "retry"
```

Comment:
We might not need this extra state here - if you inspect the current flow state, you can check if we have retried already. I believe there is a retry counter in the context.